rust/flatbuffers/src/verifier.rs

```diff
@@ -39,64 +39,141 @@ impl core::convert::AsRef<[ErrorTraceDetail]> for ErrorTrace {
     },
-    #[error("Utf8 error for string in {range:?}: {error}\n{error_trace}")]
     Utf8Error {
-        #[source]
         error: core::str::Utf8Error,
         range: Range<usize>,
         error_trace: ErrorTrace,
     },
-    #[error("String in range [{}, {}) is missing its null terminator.\n{error_trace}",
-            range.start, range.end)]
     MissingNullTerminator {
         range: Range<usize>,
         error_trace: ErrorTrace,
     },
-    #[error("Type `{unaligned_type}` at position {position} is unaligned.\n{error_trace}")]
     Unaligned {
         position: usize,
         unaligned_type: &'static str,
         error_trace: ErrorTrace,
     },
-    #[error("Range [{}, {}) is out of bounds.\n{error_trace}", range.start, range.end)]
     RangeOutOfBounds {
         range: Range<usize>,
         error_trace: ErrorTrace,
     },
-    #[error(
-        "Signed offset at position {position} has value {soffset} which points out of bounds.\
-             \n{error_trace}"
-    )]
     SignedOffsetOutOfBounds {
         soffset: SOffsetT,
         position: usize,
         error_trace: ErrorTrace,
     },
     // Dos detecting errors. These do not get error traces since it will probably be very large.
-    #[error("Too many tables.")]
     TooManyTables,
-    #[error("Apparent size too large.")]
     ApparentSizeTooLarge,
-    #[error("Nested table depth limit reached.")]
     DepthLimitReached,
 }
 
+#[cfg(any(nightly, feature = "std"))]
+impl Error for InvalidFlatbuffer {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            InvalidFlatbuffer::MissingRequiredField { .. } => None,
+            InvalidFlatbuffer::InconsistentUnion { .. } => None,
+            InvalidFlatbuffer::Utf8Error { error: source, .. } => Some(source),
+            InvalidFlatbuffer::MissingNullTerminator { .. } => None,
+            InvalidFlatbuffer::Unaligned { .. } => None,
+            InvalidFlatbuffer::RangeOutOfBounds { .. } => None,
+            InvalidFlatbuffer::SignedOffsetOutOfBounds { .. } => None,
+            InvalidFlatbuffer::TooManyTables { .. } => None,
+            InvalidFlatbuffer::ApparentSizeTooLarge { .. } => None,
+            InvalidFlatbuffer::DepthLimitReached { .. } => None,
+        }
+    }
+}
+
+impl core::fmt::Display for InvalidFlatbuffer {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        match self {
+            InvalidFlatbuffer::MissingRequiredField {
+                required,
+                error_trace,
+            } => {
+                writeln!(f, "Missing required field `{}`.\n{}", required, error_trace)?;
+            }
+            InvalidFlatbuffer::InconsistentUnion {
+                field,
+                field_type,
+                error_trace,
+            } => {
+                writeln!(f,"Union exactly one of union discriminant (`{}`) and value (`{}`) are present.\n{}", field_type, field, error_trace)?;
```

Comment:
whoops this doesn't make sense, I think it should be

"Exactly one of union discriminant (...) and value are present (...)"

Reply:
I copied the current existing string, of course I could change it but that was the existing string

Reply:
yeah I know, I think that was my typo from the original verifier implementation

Reply:
Aight, changed